### PR TITLE
[CALCITE-6678] Calcite should support dual table query when db provides this feature

### PR DESCRIPTION
**Pull request by strongduanmu**
* Use view table to implement dual table query for better optimize
* Move dual table query test case to new `dual-table-query.iq` file

### CODE REVIEW

**Review by NobiGo — APPROVED** [review]

File: core/src/test/resources/sql/dummy.iq [hunk 1]
```diff
@@ -20,58 +20,4 @@ values 1;
 EXPR$0
 1
 !ok
-
```
Comment (mihaibudiu) [inline core/src/test/resources/sql/dummy.iq:23]: Can you please add a comment here saying "do not modify this file, this is intended only for temporary debugging code"?

Comment (strongduanmu) [reply core/src/test/resources/sql/dummy.iq:23]: Good suggestion, I've added a comment.

**Review by caicancai — APPROVED** [review]

**Comment by strongduanmu** [comment]
Hi @mihaibudiu @julianhyde, could you please review this PR? I have made the changes as suggested in Jira. Thank you very much.

**Comment by mihaibudiu** [comment]
You already have an approval, the main question was whether @julianhyde is happy, he raised objections to the previous implementation. If he does not object, I will merge this in the next few days.

**Comment by mihaibudiu** [comment]
you should also squash the commits for merging.

**Pull request merged**